pkg/network/setup/netstat_test.go

```diff
@@ -302,6 +302,40 @@ var _ = Describe("netstat", func() {
 			Expect(setup.NetStat.PodInterfaceVolatileDataIsCached(setup.Vmi, primaryNetworkName)).To(BeFalse())
 			Expect(setup.NetStat.PodInterfaceVolatileDataIsCached(setup.Vmi, secondaryNetworkName)).To(BeFalse())
 		})
+
+		It("run status and expect 1 attached iface & 1 detached iface to be reported based on multus status and guest-agent data", func() {
+			Expect(
+				setup.addNetworkInterface(
+					newVMISpecIfaceWithBridgeBinding(primaryNetworkName),
+					newVMISpecPodNetwork(primaryNetworkName),
+					newDomainSpecIface(primaryNetworkName, primaryMAC),
+					primaryPodIPv4, primaryPodIPv6,
+				),
+			).To(Succeed())
+
+			setup.Vmi.Spec.Domain.Devices.Interfaces = append(setup.Vmi.Spec.Domain.Devices.Interfaces,
+				newVMISpecIfaceWithBridgeBinding(secondaryNetworkName))
+			setup.Vmi.Spec.Networks = append(setup.Vmi.Spec.Networks, newVMISpecMultusNetwork(secondaryNetworkName))
+
+			setup.addGuestAgentInterfaces(
+				newDomainStatusIface([]string{primaryGaIPv4, primaryGaIPv6}, primaryMAC, primaryIfaceName),
+			)
+			setup.Vmi.Status.Interfaces = []v1.VirtualMachineInstanceNetworkInterface{
+				{Name: primaryNetworkName, InfoSource: netvmispec.InfoSourceMultusStatus},
+				{Name: secondaryNetworkName, InfoSource: netvmispec.InfoSourceMultusStatus},
+			}
+
+			Expect(setup.NetStat.UpdateStatus(setup.Vmi, setup.Domain)).To(Succeed())
+
+			infoSourceDomainGAMultus := netvmispec.NewInfoSource(
+				netvmispec.InfoSourceDomain, netvmispec.InfoSourceGuestAgent, netvmispec.InfoSourceMultusStatus)
+			Expect(setup.Vmi.Status.Interfaces).To(Equal([]v1.VirtualMachineInstanceNetworkInterface{
+				newVMIStatusIface(primaryNetworkName, []string{primaryGaIPv4, primaryGaIPv6}, primaryMAC, primaryIfaceName, infoSourceDomainGAMultus, netsetup.DefaultInterfaceQueueCount),
+				newVMIStatusIface(secondaryNetworkName, nil, "", "", netvmispec.InfoSourceMultusStatus, 0),
+			}), "primary and secondary ifaces should exist in status, where secondary iface have multus-status only")
+
+			Expect(setup.NetStat.PodInterfaceVolatileDataIsCached(setup.Vmi, primaryNetworkName)).To(BeTrue())
```

Comment:
And what do we expect from the secondary?

Reply:
I think we should not care for the secondary because it describes a scenario it is detached.

Reply:
You have it as input in this test and I think there is a need to express what is expected of it on way or the other.

Reply:
I agree will Or, the Volatile cache is not relevant in case of detached interface. It may contain the interface or may not we don't care.

Reply:
But since I looked at his code again, it may be nice to add `setup.AddMultusOnlyInterface`:)

Reply:
I think the test needs to express the expectation explicitly for a specific scenario.
I think it is important to express what the volatile cache will be when an interface is detached.

E.g. it is not touched, it is left as before.

Reply:
Test moved outside the Conetxt() that tests volatile cache